files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md

```diff
@@ -0,0 +1,886 @@
+---
+title: "Svelte 进阶：响应性、生命周期以及无障碍"
+slug: Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_reactivity_lifecycle_accessibility
+page-type: learn-module-chapter
+---
+
+{{LearnSidebar}}
+{{PreviousMenuNext("Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_components","Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_stores", "Learn/Tools_and_testing/Client-side_JavaScript_frameworks")}}
+
+在上一篇文章中，我们为待办事项列表添加了更多功能，并开始将应用程序组织成组件。在本文中，我们将添加应用程序的最终功能，并进一步组件化我们的应用程序。我们将学习处理与更新对象和数组相关的响应性问题。为了避免常见的问题，我们需要深入了解 Svelte 的响应性系统。我们还将解决一些无障碍聚焦问题，以及其他一些问题。
+
+<table>
+  <tbody>
+    <tr>
+      <th scope="row">前提：</th>
+      <td>
+        <p>
+          至少，建议你熟悉核心<a href="/zh-CN/docs/Learn/HTML"> HTML<
+          /a>、<a 
+          href="/zh-CN/docs/Learn/CSS">CSS </a>和<a
+           href="/zh-CN/docs/Learn/JavaScript"> JavaScript </a> 语言，并且了解<a
+            href="/zh-CN/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Command_line"
+            >终端/命令行</a
+          >的使用。
+        </p>
+        <p>
+          你需要安装了 Node.js 和 npm 的终端来编译和构建应用程序。
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <th scope="row">目标：</th>
+      <td>
+        学习一些高级的 Svelte 技巧，包括解决响应性问题、与组件生命周期相关的键盘无障碍问题等等。
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+我们将重点关注涉及焦点管理的一些无障碍问题。为此，我们将利用一些访问 DOM 节点和执行 [`focus()`](/zh-CN/docs/Web/API/HTMLElement/focus) 和 [`select()`](/zh-CN/docs/Web/API/HTMLInputElement/select) 等方法的技术。我们还将学习如何声明以及清除 DOM 元素的事件监听器。
+
+我们也需要学习一些组件生命周期的知识，以了解这些 DOM 节点何时从 DOM 中挂载（mount）和卸载（unmount），以及我们如何访问它们。我们还将学习 `action` 指令，它将允许我们以可重用和声明性的方式扩展 HTML 元素的功能。
+
+最后，我们将进一步了解组件。到目前为止，我们已经看到组件如何使用属性共享数据，并使用事件和双向数据绑定与其父组件通信。现在，我们将看到组件如何公开方法和变量。
+
+本文将在课程中开发以下新组件：
+
+- `MoreActions`：显示*全选*和*移除已完成*按钮，并发出处理其功能所需的相应事件。
+- `NewTodo`：显示用于添加新待办事项的 `<input>` 字段和 *Add*（添加）按钮。
```

Comment:
Excuse me. Sometimes the reviewdog advises me to change `*` to `_`, and other times it advises me to change `_` to `*`. So what is the difference between them? @yin1999

Reply:
see: https://github.com/mdn/translated-content/discussions/12352#discussioncomment-5401774

When there are spaces or punctuations around the word, use `_`, otherwise, use `*`